active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,591 @@
+
+<!-- Use the Password property to get or set the contents of the PasswordBox. You can do this in the handler for the PasswordChanged event to perform validation while the user enters the password. Or, you can use another event, like a button Click, to perform validation after the user completes the text entry. -->
+
+Here's the XAML for a teaching tip control that demonstrates the default look of the TeachingTip with a title and subtitle. 
+
+When the user clicks the button, a teaching tip will appear and display a message to the user. 
+
+XAML
+```XAML
+<Button Content="Save" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to.">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+Here's the result when this code runs and the clicks the button.
+
+<a href="url"><img src="TeachingTipSampleApp.jpg" height="600"></a>
+
+### Add a margin 
+
+A teaching tip can have a margin added by setting the Margin property. When a teaching tip is attached to a UI element, a margin will move it away from the UI element it is pointing to. When a teaching tip is not attached to anything, a margin will move it away from the side of the app window it is positioned towards. 
+
+XAML
+```XAML
+<Button Content="Save" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            Margin="80">
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+<a href="url"><img src="TeachingTipMarginSample.jpg" height="600"></a>
+
+### Point to a UI element
+
+You can change the teaching tip to point to a specific UI element by setting the TeachingTipPlacementMode property. For a targeted teaching tip, this placement will be relative to the UI element target. For a non-targeted teaching tip, this placement will relative to app window.  
+
+XAML
+```XAML
+<Button Content="Tip Example" Click="OnFirstSaveButtonClick">
+    <muxc:TeachingTip.Attach> 
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Title="Saving automatically"
+            Subtitle="We save your changes as you go - so you never have to."
+            Margin="15"/>
+        </muxc:TeachingTip>
+    </muxc:TeachingTip.Attach>
+</Button>
+```
+
+C#
+```C#
+public void OnFirstSaveButtonClick(object sender, RoutedEventArgs args)
+{
+   AutoSaveTip.IsOpen = true;
+}
+```
+
+<a href="url"><img src="TeachingTipPointerSample.jpg" height="600"></a>
```

Comment:
I'm curious, is this a comp or a screenshot? I ask because of the padding from TT's edge to title/subtitle vs content. T StackPanel appears to render closer to the edge. While that could be easy corrected with a margin on the StackPanel, it feels odd to have this offset by default.

Reply:
These are all comps. I am hoping to replace them with screen shots when the WinUI January Prerelease becomes available.